cp23/hpi-fun2/cubes/executables/generator.cpp

```diff
@@ -72,15 +69,20 @@ vector<array<long long, 3>> generate_coords(int width, int length, int height)
                 float noise = noiseGenerator.GetNoise((float)x, (float)y, (float)z);
 
                 // If noise value is above threshold, generate a cloud block
-                if (noise > THRESHOLD)
+                if (noise > threshold)
                     coords.push_back({base_x_coordinate + x, base_y_coordinate + y, base_z_coordinate + z});
 
+                // break if we have enough coordinates
                 if (coords.size() == MAX_N)
-                    break;
+                    goto end;
             }
         }
     }
 
+end:
+    // shuffle the coordinates
+    shuffle(coords.begin(), coords.end(), default_random_engine(42));
```

Comment:
```suggestion
    shuffle(coords.begin(), coords.end());
```
`shuffle` already uses the seed set in your main method.